pkg/session/bootstrap.go

```diff
@@ -802,6 +804,12 @@ func bootstrap(s sessiontypes.Session) {
 	}
 }
 
+func getFunctionName(f func(sessiontypes.Session, int64)) string {
+	fullName := runtime.FuncForPC(reflect.ValueOf(f).Pointer()).Name()
```

Comment:
Even `getFunctionName` is used for `upgrade`s only for now, IMO it's better to add some protections like `nil` and size check. WDYT?